### PR TITLE
disable auto case indexation

### PR DESCRIPTION
**Pull request by jamal-khey**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**
this PR disable auto indexation of the cases, and intreduce a new param "indexed" to indicate wich cases should be indexed by` /v1/supervision/case/reindex`
This also PR create a supervision service to delete elk documents and reindex cases.

**What is the current behavior?**
when a case is created or duplication , this service index it automatically 

**What is the new behavior (if this is a feature change)?**
cases are indexed on demand

**Does this PR introduce a breaking change or deprecate an API?**
- [X] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**
The endpoint for reindexing all cases has been renamed from `/v1/cases/index-all` to `/v1/supervision/cases/reindex`. 
Any client applications or scripts that previously called the `/v1/cases/index-all` endpoint will need to be updated to use the new `/v1/supervision/cases/reindex` endpoint.

The database schema has also been changed to add a new column "indexed" , the values of this collumn will be set to false by default.

**Other information**:

To test this PR : 
- delete elastic search index "cases
- restart the service
- use the endpoint `/v1/supervision/cases/reindex` to recreate the index ( you should have no cases indexed , all of them are marked indexed=false by default)
- choose some cases and update "indexed" value using `PUT /cases/{caseUuid}/indexation` 
- re-do a delete index - re-create index , now the choosen cases should appear in elastic search index

### CODE REVIEW

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Jamal KHEYYAD <jamal.kheyyad at rte-international.com>
+ */
+@RunWith(SpringRunner.class)
+@AutoConfigureMockMvc
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK, properties = {"case-store-directory=/cases"})
+@ContextConfigurationWithTestChannel
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java]: done partially , the tests does not pass without `properties = {"case-store-directory=/cases"}`

File: src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            return new MockMultipartFile("file", SupervisionControllerTest.TEST_CASE, MediaType.TEXT_PLAIN_VALUE, inputStream);
+        }
+    }
+
+    @Before
+    public void setUp() {
+        fileSystem = Jimfs.newFileSystem(Configuration.unix());
+        caseService.setFileSystem(fileSystem);
+        caseService.setComputationManager(Mockito.mock(ComputationManager.class));
+        cleanDB();
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java]: done

File: src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                .andExpect(status().isOk());
+
+        String countStr = mockMvc.perform(get("/v1/supervision/cases/indexation-count"))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+        Assert.assertEquals("2", countStr);
+        Assert.assertEquals(2, supervisionService.getIndexedCaseElementsCount());
+
+    }
+
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java:100]: done

File: src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        mockMvc.perform(post("/v1/supervision/cases/reindex"))
+                .andExpect(status().isOk());
+
+        Assert.assertEquals(2, supervisionService.getIndexedCaseElementsCount());
+
+    }
+
+    @Test
+    public void testDeleteElementInfos() throws Exception {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java]: done

File: src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    private MockMvc mockMvc;
+
+    @Value("${case-store-directory}")
+    private String rootDirectory;
+
+    private static final String TEST_CASE = "testCase.xiidm";
+    private FileSystem fileSystem;
+
+    @Test
+    public void testGetElementInfosCount() throws Exception {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/test/java/com/powsybl/caseserver/SupervisionControllerTest.java]: done

File: src/main/java/com/powsybl/caseserver/CaseService.java [hunk 6]
```diff
@@ -273,6 +284,12 @@ public void disableCaseExpiration(UUID caseUuid) {
         caseMetadataEntity.setExpirationDate(null);
     }
 
+    @Transactional
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/com/powsybl/caseserver/CaseService.java]: done

File: src/main/java/com/powsybl/caseserver/CaseController.java [hunk 7]
```diff
@@ -165,9 +166,10 @@ public ResponseEntity<UUID> importCase(@RequestParam("file") MultipartFile file,
         @ApiResponse(responseCode = "500", description = "An error occurred during the case file duplication")})
     public ResponseEntity<UUID> duplicateCase(
             @RequestParam("duplicateFrom") UUID caseId,
-            @RequestParam(value = "withExpiration", required = false, defaultValue = "false") boolean withExpiration) {
+            @RequestParam(value = "withExpiration", required = false, defaultValue = "false") boolean withExpiration,
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/com/powsybl/caseserver/CaseController.java]: done

File: src/main/java/com/powsybl/caseserver/services/SupervisionService.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        AtomicReference<Long> startTime = new AtomicReference<>();
+        startTime.set(System.nanoTime());
+
+        long nbIndexesToDelete = getIndexedCaseElementsCount();
+        caseInfosRepository.deleteAll();
+        LOGGER.trace("Indexed directory elements deletion : {} seconds", TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime.get()));
+        return nbIndexesToDelete;
+    }
+
+    public long getIndexedCaseElementsCount() {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/com/powsybl/caseserver/services/SupervisionService.java]: done

File: src/main/java/com/powsybl/caseserver/CaseController.java [hunk 9]
```diff
@@ -152,9 +152,10 @@ public ResponseEntity<Boolean> exists(@PathVariable("caseUuid") UUID caseUuid) {
     @Operation(summary = "import a case")
     @SuppressWarnings("javasecurity:S5145")
     public ResponseEntity<UUID> importCase(@RequestParam("file") MultipartFile file,
-                                           @RequestParam(value = "withExpiration", required = false, defaultValue = "false") boolean withExpiration) {
+                                           @RequestParam(value = "withExpiration", required = false, defaultValue = "false") boolean withExpiration,
+                                           @RequestParam(value = "indexed", required = false, defaultValue = "false") boolean indexed) {
         LOGGER.debug("importCase request received with file = {}", file.getName());
-        UUID caseUuid = caseService.importCase(file, withExpiration);
+        UUID caseUuid = caseService.importCase(file, withExpiration, indexed);
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/com/powsybl/caseserver/CaseController.java]: done

File: src/main/java/com/powsybl/caseserver/repository/CaseMetadataRepository.java [hunk 10]
```diff
@@ -17,4 +18,8 @@
 @Repository
 public interface CaseMetadataRepository extends JpaRepository<CaseMetadataEntity, UUID> {
 
+    @Override
+    List<CaseMetadataEntity> findAllById(Iterable<UUID> uuids);
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/com/powsybl/caseserver/repository/CaseMetadataRepository.java]: done

File: src/main/java/com/powsybl/caseserver/CaseService.java [hunk 11]
```diff
@@ -238,7 +236,10 @@ UUID duplicateCase(UUID sourceCaseUuid, boolean withExpiration) {
             CaseInfos existingCaseInfos = caseInfosService.getCaseInfosByUuid(sourceCaseUuid.toString()).orElseThrow();
             CaseInfos caseInfos = createInfos(existingCaseInfos.getName(), newCaseUuid, existingCaseInfos.getFormat());
             caseInfosService.addCaseInfos(caseInfos);
-            createCaseMetadataEntity(newCaseUuid, withExpiration);
+
+            Optional<CaseMetadataEntity> existingCase = caseMetadataRepository.findById(sourceCaseUuid);
```
Comment (SlimaneAmar) [inline src/main/java/com/powsybl/caseserver/CaseService.java:240]: Throws `NO_FOUND` exception like `disableCaseExpiration`

**Comment by SlimaneAmar** [comment]
Remove any reference of the word `element`

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**